its/ruling/src/test/java/org/sonar/python/it/PythonPrAnalysisTest.java

```diff
@@ -146,7 +147,8 @@ private void assertPrAnalysisLogs(BuildResult result) {
 
     assertThat(result.getLogs())
       .contains(expectedRecomputedLog)
-      .contains(expectedRegularAnalysisLog);
+      .contains(expectedRegularAnalysisLog)
+      .contains(String.format("Cache version still up to date: \"%s\".", EXPECTED_CACHE_VERSION));
```

Comment:
This does not feel very scalable - we would have to update the constant after reach release and it does not work on the CI due to the changed build number. I would recommend to only search for a string starting with ""Cache version still up to date:". In this test we are only interested in knowing that the cache is still fresh, so we do not need the version number.